gpumon/gpumon_client.ml

```diff
@@ -17,14 +17,13 @@ open Xcp_client
 
 let xml_url () = "file:" ^ xml_path
 
-module Client = Gpumon_interface.Client(struct
-	let rpc call =
-		if !use_switch
-		then json_switch_rpc queue_name call
-		else xml_http_rpc
-                        ~srcstr:(get_user_agent ())
-                        ~dststr:"gpumon"
-                        xml_url
-                        call
-end)
-
+let rpc call =
+  if !use_switch
```

Comment:
I'd rather see `if !Xcp_service.use_switch` and no `open Xcp_client`. This module is so small that it should not need `open.

Reply:
Without opening modules:
```
let xml_url () = "file:" ^ Gpumon_interface.xml_path

let rpc call =
  if !Xcp_client.use_switch
  then Xcp_client.json_switch_rpc Gpumon_interface.queue_name call
  else Xcp_client.xml_http_rpc
      ~srcstr:(Xcp_client.get_user_agent ())
      ~dststr:"gpumon"
      xml_url
      call
module Client = Gpumon_interface.RPC_API(Idl.GenClientExnRpc(struct let rpc=rpc end))
```
I think there's a case to be made either way, what do you think?

Reply:
```ocaml
let rpc call =
  let open Xcp_client in
  if !use_switch
  then json_switch_rpc Gpumon_interface.queue_name call
  else xml_http_rpc
      ~srcstr:(get_user_agent ())
      ~dststr:"gpumon"
      xml_url
      call
```

Reply:
I think it is better to be explicit. You can still to inside `rpc`: 
```
let rpc call =
  let open Xcp_client in
```

or 

```
module C = Xcp_client
let rpc call =
  if !C.use_switch 
  then ..
```